drivers/platform/mbed/mbed_gpio_irq.cpp

```diff
@@ -0,0 +1,389 @@
+ * @param	param[in] - Configuration information for the instance.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+int32_t mbed_gpio_irq_ctrl_init(struct no_os_irq_ctrl_desc **desc,
+				const struct no_os_irq_init_param *param)
+{
+	struct no_os_irq_ctrl_desc *irq_desc;
+	struct mbed_gpio_irq_desc *mbed_gpio_irq_desc;
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	irq_desc = (struct no_os_irq_ctrl_desc *)calloc(1, sizeof(*irq_desc));
+	if (!irq_desc)
+		return -ENOMEM;
+
+	mbed_gpio_irq_desc = (struct mbed_gpio_irq_desc *)calloc(1,
+			     sizeof(*mbed_gpio_irq_desc));
+	if (!mbed_gpio_irq_desc)
+		goto err_mbed_irq_desc;
+
+	irq_desc->irq_ctrl_id = param->irq_ctrl_id;
+	irq_desc->platform_ops = &mbed_gpio_irq_ops;
+
+	/* Copy the Mbed IRQ init parameters */
+	mbed_gpio_irq_desc->gpio_irq_pin = ((struct mbed_gpio_irq_init_param *)
+					    param->extra)->gpio_irq_pin;
+
+	irq_desc->extra = mbed_gpio_irq_desc;
+	*desc = irq_desc;
+
+	return 0;
+
+err_mbed_irq_desc:
+	free(irq_desc);
+
+	return -ENOMEM;
+}
+
+/**
+ * @brief	Enable specific interrupt.
+ * @param	desc[in] - The IRQ controller descriptor.
+ * @param	irq_id[in] - Interrupt identifier.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+int32_t mbed_gpio_irq_enable(struct no_os_irq_ctrl_desc *desc, uint32_t irq_id)
+{
+	InterruptIn *ext_interrupt;
+
+	if (!desc || !desc->extra || (irq_id >= NB_GPIO_IRQS))
+		return -EINVAL;
+
+	ext_interrupt = (InterruptIn *)(((struct mbed_gpio_irq_desc *)(
+			desc->extra))->int_obj);
+	ext_interrupt->enable_irq();
+
+	return 0;
+}
+
+/**
+ * @brief	Disable specific interrupt.
+ * @param	desc[in] - The IRQ controller descriptor.
+ * @param	irq_id[in] - Interrupt identifier.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+int32_t mbed_gpio_irq_disable(struct no_os_irq_ctrl_desc *desc, uint32_t irq_id)
+{
+	InterruptIn *ext_interrupt;
+
+	if (!desc || !desc->extra || (irq_id >= NB_GPIO_IRQS))
+		return -EINVAL;
+
+	ext_interrupt = (InterruptIn *)(((struct mbed_gpio_irq_desc *)(
+			desc->extra))->int_obj);
+	ext_interrupt->disable_irq();
+
+	return 0;
+}
+
+/**
+* @brief	Registers a IRQ callback function to irq controller.
+* @param	desc[in] - The IRQ controller descriptor.
+* @param	irq_id[in] - Interrupt identifier.
+* @param	callback_desc - Descriptor of the callback. If it is NULL, the
+*			callback will be unregistered
+* @return	0 in case of success, negative error code otherwise.
+*/
+int32_t mbed_gpio_irq_register_callback(struct no_os_irq_ctrl_desc *desc,
+					uint32_t irq_id,
+					struct no_os_callback_desc *callback_desc)
+{
+	InterruptIn *ext_interrupt;
+	struct mbed_gpio_irq_desc *mbed_gpio_irq_desc;
+
+	if (!desc || !desc->extra || !callback_desc || (irq_id >= NB_GPIO_IRQS))
+		return -EINVAL;
+
+	mbed_gpio_irq_desc = (struct mbed_gpio_irq_desc *)desc->extra;
+
+	/* Store the static mbed callback functions */
+	store_mbed_callbacks();
```

Comment:
Shouldn't this be done only once, in the `init` function?

Reply:
Can be moved into init function but I guess register callback function also get called only once. Please correct if my understanding is wrong.